docs/requirements.txt

```diff
@@ -1,5 +1,7 @@
 doc8>=0.8.0
 furo>=2021.10.09
 readme-renderer>=17.4
-Sphinx>=4.3.1
+# Remove this upper bound when twine's minimum Python is 3.9+.
+# See: https://github.com/sphinx-doc/sphinx/issues/11767
+Sphinx>=6,<7.1
```

Comment:
NB: This is my temporary hack for working around https://github.com/sphinx-doc/sphinx/issues/11767.

Another option here would be to do the docs build in CI on Python 3.9 or newer, since those don't seem to be affected. But I wasn't sure if `tox -e docs` was expected to work on all supported Python versions or not 🙂